aviatrix/resource_aviatrix_spoke_external_device_conn.go

```diff
@@ -382,6 +382,43 @@ func resourceAviatrixSpokeExternalDeviceConn() *schema.Resource {
 				ForceNew:    true,
 				Description: "Backup Local LAN IP.",
 			},
+			"enable_bfd": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     false,
+				Description: "Enable BGP BFD connection.",
+			},
+			"bgp_bfd": {
+				Type:        schema.TypeList,
+				Optional:    true,
+				Description: "BGP BFD configuration details applied to a BGP session.",
+				MaxItems:    1,
```

Comment:
Why is this a list with max one? Could it be an optional Map instead? The code that extract this doesn't seem to care about the MaxItems limit?

Reply:
I wanted to provide support for bgp_bfd config that looks like this
```
bgp_bfd {
     transmit_interval = 300
     receive_interval   = 300
     multiplier              = 3
}
```
TF map does not support complex key - values which in our case bgp_bfd value is. Hence the choice was made to select the list type with a max item list of 1. For a given external connection there will only be 1 bgp_bfd config.
This implementation is an extension of this merged PR - https://github.com/AviatrixSystems/terraform-provider-aviatrix/pull/2051